tests/test_model_selection/test_timegapsplit.py

```diff
@@ -55,6 +60,55 @@ def test_timegapsplit_too_big_gap():
         print("Successfully failed")
 
 
+def test_timegapsplit_using_splits():
+    cv = TimeGapSplit(
+        date_serie=df["date"],
+        train_duration=timedelta(days=5),
+        valid_duration=timedelta(days=3),
+        gap_duration=timedelta(days=1),
+        n_splits=3
+    )
+    assert len(list(cv.split(X_train, y_train))) == 3
+
+
+@pytest.mark.xfail(raises=ValueError, strict=True)
+def test_timegapsplit_too_many_splits():
+    cv = TimeGapSplit(
+        date_serie=df["date"],
+        train_duration=timedelta(days=5),
+        valid_duration=timedelta(days=3),
+        gap_duration=timedelta(days=1),
+        n_splits=7
+    )
+    list(cv.split(X_train, y_train))
+
+
+@pytest.mark.xfail(raises=ValueError, strict=True)
```

Comment:
I have never used `xfail` before. I've just checked the documentation but I wonder; what is the main benefit of using this as opposed to using this:

```python
def test_timegapsplit_train_or_nsplit():
   with pytest.raises(ValueError):
	    cv = TimeGapSplit(
	        date_serie=df["date"],
	        train_duration=None,
	        valid_duration=timedelta(days=3),
	        gap_duration=timedelta(days=5),
	        n_splits=None
	    )
	    cv.split(X_train, y_train)
```